📘 Notebook Cell 1 — Markdown
Title: Understanding the Old NANP Telephone Number Format

The goal of this step is to restate the structure of the old North American Numbering Plan (NANP) and identify how many choices each digit position allows.
We will break the telephone number into:

Area code: NYX

Office code: NNX

Station code: XXXX

Where:

N = digits 2–9 (8 possibilities)

Y = digits 0–1 (2 possibilities)

X = digits 0–9 (10 possibilities)

We will compute the number of possible codes for each part.

In [3]:
# Number of possibilities for each digit type
N = 8   # digits 2–9
Y = 2   # digits 0–1
X = 10  # digits 0–9

# Old plan formats
area_codes_old = N * Y * X
office_codes_old = N * N * X
station_codes = X ** 4

area_codes_old, office_codes_old, station_codes


(160, 640, 10000)

📘 Notebook Cell 3 — Markdown
Explanation of the Computation

Area code (NYX)

First digit: N → 8 choices

Second digit: Y → 2 choices

Third digit: X → 10 choices

Total: 
8
⋅
2
⋅
10
=
160

Office code (NNX)

First digit: N → 8 choices

Second digit: N → 8 choices

Third digit: X → 10 choices

Total: 
8
⋅
8
⋅
10
=
640

Station code (XXXX)

Four digits, each with 10 choices

Total: 
10
4
=
10,000

Next, we will compute the total number of possible telephone numbers under the old plan.

In [4]:
# Total possible telephone numbers under the old plan
total_old_plan = area_codes_old * office_codes_old * station_codes
total_old_plan


1024000000

📘 Notebook Cell 5 — Markdown
Final Explanation and Result

To find the total number of possible telephone numbers under the old plan, we multiply:

Number of area codes: 160

Number of office codes: 640

Number of station codes: 10,000

So the total number of possible telephone numbers is:

160
×
640
×
10,000
=
1,024,000,000
There were 1.024 billion possible telephone numbers under the old NANP system.